plutus-ledger-api/src/PlutusLedgerApi/Common/SerialisedScript.hs

```diff
@@ -59,6 +61,14 @@ implies that it is (almost certainly) an encoded script and the script does not
 
 Note: Parameterized over the ledger-plutus-version since the builtins allowed (during decoding) differs.
 -}
-isScriptWellFormed :: LedgerPlutusVersion -> ProtocolVersion -> SerialisedScript -> Bool
-isScriptWellFormed lv pv = isRight . CBOR.deserialiseFromBytes (scriptCBORDecoder lv pv) . fromStrict . fromShort
-
+assertScriptWellFormed :: MonadError CBOR.DeserialiseFailure m
+                       => LedgerPlutusVersion
+                       -> ProtocolVersion
+                       -> SerialisedScript
+                       -> m ()
+assertScriptWellFormed lv pv = fromShort
+                             >>> fromStrict
+                             >>> CBOR.deserialiseFromBytes (scriptCBORDecoder lv pv)
+                             -- throw away the success result
+                             >>> second (const ())
```

Comment:
I could refrain from throwing away the result and just return `Right ScriptForExecution`: this would let me call this function inside `mkTermToEvaluate` , thus saving on duplicated code. 
I didn't do it because I was worried that I would be exposing `ScriptForExecution` to the api. I am listening for your opinions

Reply:
Also I didn't go wiht the isomorphic `Maybe DeserialiseFailure` because that's just plain confusing

Reply:
I agree with your judgements here :+1: but maybe write it down for future reference?